src/Common.cpp

```diff
@@ -43,6 +44,23 @@ int Common::RandomInt(int min, int max) {
     return dist(rng);
 }
 
+template <typename T>
+T Common::RandomEvent(int denominator, std::vector<int> numerators, std::vector<T> outcomes) {
+    std::vector <int> events;
+    for (auto it : numerators)
+    // for (std::vector <int>::iterator it = numerators.begin(); it != numerators.end(); it++)
+    {
+        for (int i = 0; i < it; i++)
+        {
+            events.push_back(it);
+        }
+    }
+
+    int rIndex = Common::RandomInt(0, denominator);
+    int outIndex = events.at(rIndex);
+    return outcomes.at(outIndex);
+}
+
```

Comment:
This does not work, please test it. I tested `RandomEvent<int>(8, {2,2,2,2}, {100, 101, 102, 103})` and it either always returns 102 or throws an out of range exception.

Reply:
I believe I have resolved that. I'm afraid I haven't been able to test it because I encountered some type of linking errors when I tried to do that and wasn't able to resolve them

Reply:
What linking problems are you having? Did you follow the instructions in the readme?

Reply:
I tried to but I couldn't use them like that. I used the cmake extension from VS code and it worked in compiling the game as is. When I tried to replace the `RandomInt()` function with `RandomEvent()` (with appropriate params, of course) it gave me `LNK 2019 unresolved external symbol error`. Admittedly, I don't have experience with CMake, especially on Windows on which I working now

Reply:
Why replace `RandomInt()` with `RandomEvent()`? It should be a brand new function that doesn't replace anything else. Try building from the command line, like in the instructions, that might fix it.

Anyway, the build check has failed so it's still not working so far, but if you can't fix your linking problems it you can always test it separately by making a new cpp file and building & running it there. Then you can paste the function body here :)

Reply:
It's working now. I solved the linking error, it was due to something related to templates and I also fixed the `out_of_bounds` issue which was from a typo. Tested as well